qiskit/transpiler/target.py

```diff
@@ -0,0 +1,600 @@
+    def update_from_instruction_schedule_map(self, inst_map, inst_name_map=None, error_dict=None):
+        """Update the target from an instruction schedule map.
+
+        If the input instruction schedule map contains new instructions not in
+        the target they will be added. However if it contains additional qargs
+        for an existing instruction in the target it will error.
+
+        Args:
+            inst_map (InstructionScheduleMap): The instruction
+            inst_name_map (dict): An optional dictionary that maps any
+                instruction name in ``inst_map`` to an instruction object
+            error_dict (dict): A dictionary of errors of the form::
+
+                {gate_name: {qarg: error}}
+
+            for example::
+
+                {'rx': {(0, ): 1.4e-4, (1, ): 1.2e-4}}
+
+            For each entry in the ``inst_map`` if ``error_dict`` is defined
+            a when updating the ``Target`` the error value will be pulled from
+            this dictionary. If one is not found in ``error_dict`` then
+            ``None`` will be used.
+
+        Raises:
+            ValueError: If ``inst_map`` contains new instructions and
+                ``inst_name_map`` isn't specified
+            KeyError: If a ``inst_map`` contains a qarg for an instruction
+                that's not in the target
+        """
+        for inst in inst_map.instructions:
+            out_props = {}
+            for qarg in inst_map.qubits_with_instruction(inst):
+                sched = inst_map.get(inst, qarg)
+                val = InstructionProperties(schedule=sched)
+                try:
+                    qarg = tuple(qarg)
+                except TypeError:
+                    qarg = (qarg,)
+                if inst in self._gate_map:
+                    if self.dt is not None:
+                        val.length = sched.duration * self.dt
+                    else:
+                        val.length = None
+                    if error_dict is not None:
+                        error_inst = error_dict.get(inst)
+                        if error_inst:
+                            error = error_inst.get(qarg)
+                            val.error = error
+                        else:
+                            val.error = None
+                    else:
+                        val.error = None
+                out_props[qarg] = val
+            if inst not in self._gate_map:
+                if inst_name_map is not None:
+                    self.add_instruction(inst_name_map[inst], out_props, name=inst)
+                else:
+                    raise ValueError(
+                        "An inst_name_map kwarg must be specified to add new "
+                        "instructions from an InstructionScheduleMap"
+                    )
+            else:
+                for qarg, prop in out_props.items():
+                    self.update_instruction_properties(inst, qarg, prop)
+
+    @property
+    def qargs(self):
```

Comment:
It's more if you wanted to build an interaction map for the device (thinking like for a `coupling_map` but >2q). I don't think there is any specific use case I had in mind for right now, but it seemed like something people might need. But, I can remove it if you'd prefer

Reply:
I see the possible utility, but it doesn't seem likely it would be especially clear what this is. Do we layout a definition for what a `qarg` is in the `Target` documentation?

I'd probably err toward not including this in favor of pointing to a structure that links between a set of `qargs` and the `operation` type. (Knowing that I have an `operation` between `qargs` 34, 35, and 39 isn't likely to be very useful without knowing what that `operation` is.)

Reply:
So there was a use case for this already, in #7227 I used this to figure out if any gates aren't on all qubits. It was needed for everything >1q (for 1q gates I could figure it from a simple range).